### PR TITLE
Enable -Wconversion in src/platform/Darwin.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
Also fixes incorrect handling of the bssid from CWNetwork.

### CODE REVIEW

**Comment by bzbarsky-apple** [comment]
This may need https://github.com/project-chip/connectedhomeip/pull/25324 to merge before it will compile...

**Review by woody-apple — APPROVED** [review]

**Review by yunhanw-google — APPROVED** [review]

**Review by jtung-apple — APPROVED** [review]

File: src/platform/Darwin/WiFi/NetworkCommissioningWiFiDriver.mm [hunk 1]
```diff
@@ -72,18 +73,23 @@ bool Next(WiFiScanResponse & scanResponse) override
...
+        destination.rssi = static_cast<int8_t>(source.rssiValue);
+
+        NSData * ssidData = source.ssidData;
+        destination.ssidLen = static_cast<uint8_t>(std::min(ssidData.length, DeviceLayer::Internal::kMaxWiFiSSIDLength));
+        memcpy(destination.ssid, ssidData.bytes, destination.ssidLen);
+
+        // source.bssid looks like "00:00:00:00:00:00" if it's not nil.
+        NSString * bssid = source.bssid;
+        // 3 chars per byte, except the last byte.
+        if (bssid.length == 3 * sizeof(destination.bssid) - 1) {
```
Comment (msandstedt) [inline src/platform/Darwin/WiFi/NetworkCommissioningWiFiDriver.mm:87]: Should an error be logged if the bssid isn't nil, but also ins't the right length?

**Review by msandstedt — APPROVED** [review]

File: src/platform/Darwin/WiFi/NetworkCommissioningWiFiDriver.mm (hunk 1 above)
Comment (bzbarsky-apple) [reply src/platform/Darwin/WiFi/NetworkCommissioningWiFiDriver.mm:87]: Probably a good idea...  In practice, it's always going to be nil, I suspect.

Comment (bzbarsky-apple) [reply src/platform/Darwin/WiFi/NetworkCommissioningWiFiDriver.mm:87]: Filed https://github.com/project-chip/connectedhomeip/issues/25361 to do that.

**Pull request merged**